server/libs/aws/cloudwatch.js

```diff
@@ -114,54 +113,82 @@ export default aws => {
         .then(() => initRuleTarget(cloudWatchEvents, ruleName))
     },
 
+    /**
+     * Fetch logs based on a CloudWatch stream name
+     * 
+     * @param {string} logStreamName - CloudWatch stream to read
+     * @param {boolean} truncate - limit results to 1000 lines
+     * @param {function} callback - Callback each time logs are received 
+     * 
+     * @returns {Promise} - Resolves when all logs have been retrieved
+     */
+    getLogs(logStreamName, truncate, callback) {
+      return new Promise((resolve, reject) => {
```

Comment:
Since we are going to do an implementation it might be worth making the getCloudwatchLogs non-recursive. This might help a little with confusion about about passing a callback, resolve and reject functions to the method.